tests/datasets/test_streaming.py

```diff
@@ -224,33 +224,43 @@ def test_dataloader_single_device(remote_local: Tuple[str, str], batch_size: int
 @pytest.mark.world_size(2)
 @pytest.mark.parametrize("batch_size", [4])
 @pytest.mark.parametrize("drop_last", [False, True])
+@pytest.mark.parametrize("multinode", [False, True])
 @pytest.mark.parametrize("num_samples", [30, 31])
 @pytest.mark.parametrize("num_workers", [1, 3])
 @pytest.mark.parametrize("shuffle", [False, True])
-def test_dataloader_multi_device(remote_local: Tuple[str, str], batch_size: int, drop_last: bool, num_samples: int,
-                                 num_workers: int, shuffle: bool):
+def test_dataloader_multi_device(remote_local: Tuple[str, str], batch_size: int, drop_last: bool, multinode: bool,
+                                 num_samples: int, num_workers: int, shuffle: bool):
+
+    if multinode:
+        # Force different nodes
+        os.environ["LOCAL_RANK"] = str(0)
```

Comment:
Curious if this test works? The distributed is configured before any individual test runs, and the dist helpers should complain if there is an environment variable mismatch:

https://github.com/mosaicml/composer/blob/0f1624ed3d36aa7c032209f115b3af9ec9e0402d/composer/utils/dist.py#L77-L80

I see it's marked as `@daily`, but curious if you tried to run it locally with any luck?

Reply:
In addition, is it possible to use the `monkeypatch` fixture instead of setting these variables directly? `monkeypatch` will reset the variables back to their previous values after the test runs, so any other tests in the same process will have the original values.

Reply:
Changing the `os.environ` does seem to work.. and I definitely saw different behavior with `multinode` on vs. off as I was debugging. I think maybe becasue I'm only changing the local rank and node rank, `dist` isn't complaining (world size is still 2). I will try to use the `monkeypatch` fixture as suggested.

Reply:
Yeah so to close the loop here on why that `RuntimeError` does not occur, it's because the vars I'm changing don't have an associated `fetch_fn_name` which is what `composer.utils.dist` requires to run that code block you posted .